internal/dag/dag.go

```diff
@@ -557,12 +559,13 @@ type Vertex interface {
 	Visit(func(Vertex))
 }
 
-// Secret represents a K8s Sevice as a DAG vertex. A Serivce is
+// Service represents a K8s Sevice as a DAG vertex. A Service is
 // a leaf in the DAG.
 type Service struct {
 	object *v1.Service
 
 	*v1.ServicePort
+	Weight *int
```

Comment:
I suggest making this an int, not a *int, then we don't need to deal with nil fields and we can define what a weight of zero means; we can say it means unset. If someone references a service with a weight of zero in the ingressroute document, we can filter it out during dag build time. (but see my comments above about this field)